addons/test_mail/tests/test_mail_activity.py

```diff
@@ -215,6 +231,150 @@ def test_action_feedback_attachment(self):
             self.assertEqual(attachment.res_id, activity_message.id)
             self.assertEqual(attachment.res_model, activity_message._name)
 
+    def test_mail_activity_state(self):
+        today_utc = datetime(2020, 1, 1, 16, 0, 0)
+
+        class MockedDatetime(datetime):
+            @classmethod
+            def utcnow(cls):
+                return today_utc
+
+        with patch('odoo.addons.mail.models.mail_activity.datetime', MockedDatetime):
+            activity_1 = self.env['mail.activity'].create({
+                'summary': 'Test',
+                'activity_type_id': 1,
+                'res_model_id': self.env.ref('base.model_res_partner').id,
+                'res_id': self.partner_admin.id,
+                'date_deadline': today_utc,
+                'user_id': self.user_utc.id,
+            })
+
+            activity_2 = activity_1.copy()
+            activity_2.user_id = self.user_australia
+            activity_3 = activity_1.copy()
+            activity_3.date_deadline += relativedelta(hours=8)
+
+            self.assertEqual(activity_1.state, 'today')
+            self.assertEqual(activity_2.state, 'overdue')
+            self.assertEqual(activity_3.state, 'today')
+
+    def test_mail_activity_mixin_search_state_1(self):
+        today_utc = datetime(2020, 1, 1, 16, 0, 0)
+
+        class MockedDatetime(datetime):
+            @classmethod
+            def utcnow(cls):
+                return today_utc
+
+        origin_1 = self.partner_admin
+        origin_2 = self.partner_employee
+
+        # Create some partners without activity schedule on it for testing
+        for partner_i in range(10):
```

Comment:
Don't use partner model for tests. You have plenty of test models available in test_mail (mail.test.activity notably).